src/models/devicemodel.cpp

```diff
@@ -51,6 +51,9 @@ bool DeviceModel::fromJson(const Keys* keys, const QByteArray& s) {
   }
 
   m_rawJson = s;
+#if !defined(UNIT_TEST)
+  writeSettings();
```

Comment:
As this is now a side effect of reading the device model from the Json, the previous code that used writeSettings() in the caller appears to be a better design.

AFAIK, DeviceModel::fromJson is called from MozillaVPN::completeAuthentication (with the list of devices without the current device) and MozillaVPN::accountChecked (with the list of devices including the current device), and both of them called writeSettings. So, I don't understand the cause of the bug. Could you please clarify if I'm missing something?